tmate-ssh-server.c

```diff
@@ -267,9 +270,24 @@ static void client_bootstrap(struct tmate_session *_session)
 	register_on_ssh_read(client);
 
 	switch (client->role) {
-	case TMATE_ROLE_DAEMON:		tmate_spawn_daemon(_session);		break;
-	case TMATE_ROLE_PTY_CLIENT:	tmate_spawn_pty_client(_session);	break;
-	case TMATE_ROLE_EXEC:		tmate_spawn_exec(_session);		break;
+	case TMATE_ROLE_DAEMON:
+      if (handle_role_daemon)
+        tmate_spawn_daemon(_session);
+      else
+        tmate_info("Ignoring daemon connection");
+      break;
+	case TMATE_ROLE_PTY_CLIENT:
+      if (handle_role_pty_client)
+        tmate_spawn_pty_client(_session);
+      else
+        tmate_info("Ignoring client connection");
+      break;
+	case TMATE_ROLE_EXEC:
```

Comment:
As suggested in my previous comment, `EXEC` should be handled when requested by an ssh client.

Reply:
Fixed. I tested with `exec` and it does not work but for a different reason. It fails in 
```
if (ssh_event_dopoll(mainloop, -1) == SSH_ERROR)
			tmate_fatal_quiet("Error polling ssh socket: %s", ssh_get_error(session));
}
``` 

I didn't pursue further for now because I'm not sure if `exec` is important in the workflow we want.

Reply:
(the behavior is the same on the HEAD of the master branch, so it's not a regression introduced by the changes here)